include/SugarEmailAddress/SugarEmailAddress.php

```diff
@@ -1144,21 +1193,50 @@ public function AddUpdateEmailAddress($addr, $invalid = 0, $opt_out = 0, $id = n
             }
         }
 
+        // confirmed opt in check
+        $isValidEmailAddress = ($opt_out !== 1 && $invalid !== 1);
+        $this->retrieve($id);
+        $optInIndication = $this->getOptInIndication();
+        if (
+           $isValidEmailAddress
+           && EmailAddressIndicator::isOptedInStatus($optInIndication)
```

Comment:
You using this static function only in this class. Why did not make a new private method here? Why we need a brand new class for it?

Reply:
Good point. I was having trouble deciding where this should go, as I was in the process of moving the functionality in the Basic class to the SugarEmailAddress class. So I will refactor out the duplicate function in the basic class and move this method into the SugarEmailAddress. Thus removing the need for the static method.